### PR TITLE
Migrate `logs` resources to AWS SDK for Go v2 

### PR DESCRIPTION
**Pull request by ewbankkit**
### Description

Ibid.

### Output from Acceptance Testing

```console
% make testacc TESTARGS='-run=TestAccLogs' PKG=logs ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/logs/... -v -count 1 -parallel 3  -run=TestAccLogs -timeout 360m
=== RUN   TestAccLogsDataProtectionPolicyDocumentDataSource_basic
=== PAUSE TestAccLogsDataProtectionPolicyDocumentDataSource_basic
=== RUN   TestAccLogsDataProtectionPolicyDocumentDataSource_empty
=== PAUSE TestAccLogsDataProtectionPolicyDocumentDataSource_empty
=== RUN   TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements
=== PAUSE TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements
=== RUN   TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation
=== PAUSE TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation
=== RUN   TestAccLogsDataProtectionPolicy_basic
=== PAUSE TestAccLogsDataProtectionPolicy_basic
=== RUN   TestAccLogsDataProtectionPolicy_disappears
=== PAUSE TestAccLogsDataProtectionPolicy_disappears
=== RUN   TestAccLogsDataProtectionPolicy_policyDocument
=== PAUSE TestAccLogsDataProtectionPolicy_policyDocument
=== RUN   TestAccLogsDestinationPolicy_basic
=== PAUSE TestAccLogsDestinationPolicy_basic
=== RUN   TestAccLogsDestination_basic
=== PAUSE TestAccLogsDestination_basic
=== RUN   TestAccLogsDestination_disappears
=== PAUSE TestAccLogsDestination_disappears
=== RUN   TestAccLogsDestination_tags
=== PAUSE TestAccLogsDestination_tags
=== RUN   TestAccLogsDestination_update
=== PAUSE TestAccLogsDestination_update
=== RUN   TestAccLogsGroupDataSource_basic
=== PAUSE TestAccLogsGroupDataSource_basic
=== RUN   TestAccLogsGroup_basic
=== PAUSE TestAccLogsGroup_basic
=== RUN   TestAccLogsGroup_nameGenerate
=== PAUSE TestAccLogsGroup_nameGenerate
=== RUN   TestAccLogsGroup_namePrefix
=== PAUSE TestAccLogsGroup_namePrefix
=== RUN   TestAccLogsGroup_disappears
=== PAUSE TestAccLogsGroup_disappears
=== RUN   TestAccLogsGroup_tags
=== PAUSE TestAccLogsGroup_tags
=== RUN   TestAccLogsGroup_kmsKey
=== PAUSE TestAccLogsGroup_kmsKey
=== RUN   TestAccLogsGroup_retentionPolicy
=== PAUSE TestAccLogsGroup_retentionPolicy
=== RUN   TestAccLogsGroup_multiple
=== PAUSE TestAccLogsGroup_multiple
=== RUN   TestAccLogsGroup_skipDestroy
=== PAUSE TestAccLogsGroup_skipDestroy
=== RUN   TestAccLogsGroupsDataSource_basic
=== PAUSE TestAccLogsGroupsDataSource_basic
=== RUN   TestAccLogsGroupsDataSource_noPrefix
=== PAUSE TestAccLogsGroupsDataSource_noPrefix
=== RUN   TestAccLogsMetricFilter_basic
=== PAUSE TestAccLogsMetricFilter_basic
=== RUN   TestAccLogsMetricFilter_disappears
=== PAUSE TestAccLogsMetricFilter_disappears
=== RUN   TestAccLogsMetricFilter_Disappears_logGroup
=== PAUSE TestAccLogsMetricFilter_Disappears_logGroup
=== RUN   TestAccLogsMetricFilter_many
=== PAUSE TestAccLogsMetricFilter_many
=== RUN   TestAccLogsMetricFilter_update
=== PAUSE TestAccLogsMetricFilter_update
=== RUN   TestAccLogsQueryDefinition_basic
=== PAUSE TestAccLogsQueryDefinition_basic
=== RUN   TestAccLogsQueryDefinition_disappears
=== PAUSE TestAccLogsQueryDefinition_disappears
=== RUN   TestAccLogsQueryDefinition_rename
=== PAUSE TestAccLogsQueryDefinition_rename
=== RUN   TestAccLogsQueryDefinition_logGroups
=== PAUSE TestAccLogsQueryDefinition_logGroups
=== RUN   TestAccLogsResourcePolicy_basic
=== PAUSE TestAccLogsResourcePolicy_basic
=== RUN   TestAccLogsResourcePolicy_ignoreEquivalent
=== PAUSE TestAccLogsResourcePolicy_ignoreEquivalent
=== RUN   TestAccLogsStream_basic
=== PAUSE TestAccLogsStream_basic
=== RUN   TestAccLogsStream_disappears
=== PAUSE TestAccLogsStream_disappears
=== RUN   TestAccLogsStream_Disappears_logGroup
=== PAUSE TestAccLogsStream_Disappears_logGroup
=== RUN   TestAccLogsSubscriptionFilter_basic
=== PAUSE TestAccLogsSubscriptionFilter_basic
=== RUN   TestAccLogsSubscriptionFilter_many
=== PAUSE TestAccLogsSubscriptionFilter_many
=== RUN   TestAccLogsSubscriptionFilter_disappears
=== PAUSE TestAccLogsSubscriptionFilter_disappears
=== RUN   TestAccLogsSubscriptionFilter_Disappears_logGroup
=== PAUSE TestAccLogsSubscriptionFilter_Disappears_logGroup
=== RUN   TestAccLogsSubscriptionFilter_DestinationARN_kinesisDataFirehose
=== PAUSE TestAccLogsSubscriptionFilter_DestinationARN_kinesisDataFirehose
=== RUN   TestAccLogsSubscriptionFilter_DestinationARN_kinesisStream
=== PAUSE TestAccLogsSubscriptionFilter_DestinationARN_kinesisStream
=== RUN   TestAccLogsSubscriptionFilter_distribution
=== PAUSE TestAccLogsSubscriptionFilter_distribution
=== RUN   TestAccLogsSubscriptionFilter_roleARN
=== PAUSE TestAccLogsSubscriptionFilter_roleARN
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_basic
=== CONT  TestAccLogsGroupsDataSource_noPrefix
=== CONT  TestAccLogsGroupDataSource_basic
--- PASS: TestAccLogsGroupsDataSource_noPrefix (26.33s)
=== CONT  TestAccLogsDataProtectionPolicy_policyDocument
--- PASS: TestAccLogsGroupDataSource_basic (26.40s)
=== CONT  TestAccLogsDestination_update
--- PASS: TestAccLogsDataProtectionPolicy_policyDocument (66.03s)
=== CONT  TestAccLogsDestination_tags
--- PASS: TestAccLogsDestination_update (169.76s)
=== CONT  TestAccLogsDestination_disappears
--- PASS: TestAccLogsDestination_tags (106.24s)
=== CONT  TestAccLogsDestination_basic
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_basic (205.45s)
=== CONT  TestAccLogsDestinationPolicy_basic
--- PASS: TestAccLogsDestination_disappears (64.20s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation (3.27s)
=== CONT  TestAccLogsDataProtectionPolicy_disappears
--- PASS: TestAccLogsDestination_basic (68.34s)
=== CONT  TestAccLogsDataProtectionPolicy_basic
--- PASS: TestAccLogsDestinationPolicy_basic (88.84s)
=== CONT  TestAccLogsGroup_kmsKey
--- PASS: TestAccLogsDataProtectionPolicy_basic (32.69s)
=== CONT  TestAccLogsGroupsDataSource_basic
--- PASS: TestAccLogsDataProtectionPolicy_disappears (46.37s)
=== CONT  TestAccLogsGroup_skipDestroy
--- PASS: TestAccLogsGroupsDataSource_basic (22.93s)
=== CONT  TestAccLogsGroup_multiple
--- PASS: TestAccLogsGroup_skipDestroy (23.73s)
=== CONT  TestAccLogsGroup_retentionPolicy
--- PASS: TestAccLogsGroup_multiple (25.38s)
=== CONT  TestAccLogsGroup_tags
--- PASS: TestAccLogsGroup_kmsKey (75.09s)
=== CONT  TestAccLogsGroup_disappears
--- PASS: TestAccLogsGroup_disappears (21.79s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements (3.98s)
=== CONT  TestAccLogsGroup_namePrefix
--- PASS: TestAccLogsGroup_retentionPolicy (71.20s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_empty
--- PASS: TestAccLogsGroup_tags (71.87s)
=== CONT  TestAccLogsStream_basic
--- PASS: TestAccLogsGroup_namePrefix (32.40s)
=== CONT  TestAccLogsSubscriptionFilter_roleARN
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_empty (27.18s)
=== CONT  TestAccLogsSubscriptionFilter_distribution
--- PASS: TestAccLogsStream_basic (28.57s)
=== CONT  TestAccLogsSubscriptionFilter_DestinationARN_kinesisStream
--- PASS: TestAccLogsSubscriptionFilter_distribution (61.21s)
=== CONT  TestAccLogsSubscriptionFilter_DestinationARN_kinesisDataFirehose
--- PASS: TestAccLogsSubscriptionFilter_DestinationARN_kinesisStream (66.45s)
=== CONT  TestAccLogsSubscriptionFilter_Disappears_logGroup
--- PASS: TestAccLogsSubscriptionFilter_roleARN (94.03s)
=== CONT  TestAccLogsSubscriptionFilter_disappears
--- PASS: TestAccLogsSubscriptionFilter_Disappears_logGroup (36.18s)
=== CONT  TestAccLogsSubscriptionFilter_many
--- PASS: TestAccLogsSubscriptionFilter_disappears (36.04s)
=== CONT  TestAccLogsSubscriptionFilter_basic
--- PASS: TestAccLogsSubscriptionFilter_DestinationARN_kinesisDataFirehose (85.45s)
=== CONT  TestAccLogsStream_Disappears_logGroup
--- PASS: TestAccLogsSubscriptionFilter_many (46.55s)
=== CONT  TestAccLogsStream_disappears
--- PASS: TestAccLogsStream_Disappears_logGroup (19.09s)
=== CONT  TestAccLogsQueryDefinition_basic
--- PASS: TestAccLogsSubscriptionFilter_basic (47.89s)
=== CONT  TestAccLogsQueryDefinition_rename
--- PASS: TestAccLogsStream_disappears (23.51s)
=== CONT  TestAccLogsQueryDefinition_disappears
--- PASS: TestAccLogsQueryDefinition_basic (26.76s)
=== CONT  TestAccLogsGroup_nameGenerate
=== CONT  TestAccLogsMetricFilter_disappears
--- PASS: TestAccLogsQueryDefinition_disappears (20.36s)
--- PASS: TestAccLogsQueryDefinition_rename (44.97s)
=== CONT  TestAccLogsGroup_basic
--- PASS: TestAccLogsGroup_nameGenerate (26.29s)
=== CONT  TestAccLogsMetricFilter_update
--- PASS: TestAccLogsMetricFilter_disappears (23.68s)
=== CONT  TestAccLogsMetricFilter_basic
--- PASS: TestAccLogsGroup_basic (26.78s)
=== CONT  TestAccLogsMetricFilter_many
--- PASS: TestAccLogsMetricFilter_basic (26.92s)
=== CONT  TestAccLogsQueryDefinition_logGroups
--- PASS: TestAccLogsMetricFilter_update (45.78s)
=== CONT  TestAccLogsResourcePolicy_ignoreEquivalent
--- PASS: TestAccLogsMetricFilter_many (38.47s)
=== CONT  TestAccLogsResourcePolicy_basic
--- PASS: TestAccLogsResourcePolicy_ignoreEquivalent (37.31s)
=== CONT  TestAccLogsMetricFilter_Disappears_logGroup
--- PASS: TestAccLogsQueryDefinition_logGroups (46.28s)
--- PASS: TestAccLogsMetricFilter_Disappears_logGroup (19.03s)
--- PASS: TestAccLogsResourcePolicy_basic (42.03s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/logs	763.236s
```

### CODE REVIEW

File: internal/acctest/vcr.go [hunk 1]
```diff
@@ -276,12 +273,12 @@ func vcrProviderConfigureContextFunc(provider *schema.Provider, configureContext
...
-			if errs.Contains(r.Error, cassette.ErrInteractionNotFound.Error()) {
-				r.Retryable = aws.Bool(false)
-			}
-		})
+		// meta.LogsConn(ctx).Handlers.AfterRetry.PushFront(func(r *request.Request) {
+		// 	// We have to use 'Contains' rather than 'errors.Is' because 'awserr.Error' doesn't implement 'Unwrap'.
+		// 	if errs.Contains(r.Error, cassette.ErrInteractionNotFound.Error()) {
+		// 		r.Retryable = aws.Bool(false)
+		// 	}
+		// })
```
Comment (jar-b) [inline internal/acctest/vcr.go:281]: Can this be permanently removed now?

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc TESTARGS='-run=TestAccLogs' PKG=logs
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/logs/... -v -count 1 -parallel 20  -run=TestAccLogs -timeout 360m

--- PASS: TestAccLogsGroup_skipDestroy (165.66s)
=== CONT  TestAccLogsSubscriptionFilter_basic
--- PASS: TestAccLogsGroup_multiple (183.58s)
=== CONT  TestAccLogsSubscriptionFilter_disappears
--- PASS: TestAccLogsGroupDataSource_basic (248.21s)
=== CONT  TestAccLogsSubscriptionFilter_many
--- PASS: TestAccLogsGroupsDataSource_noPrefix (261.18s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation
--- PASS: TestAccLogsGroupsDataSource_basic (271.30s)
=== CONT  TestAccLogsDataProtectionPolicy_disappears
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnNoOperation (18.49s)
=== CONT  TestAccLogsDataProtectionPolicy_basic
--- PASS: TestAccLogsStream_basic (284.40s)
=== CONT  TestAccLogsQueryDefinition_basic
--- PASS: TestAccLogsGroup_retentionPolicy (422.31s)
=== CONT  TestAccLogsResourcePolicy_ignoreEquivalent
--- PASS: TestAccLogsQueryDefinition_basic (241.12s)
=== CONT  TestAccLogsResourcePolicy_basic
--- PASS: TestAccLogsGroup_kmsKey (758.50s)
=== CONT  TestAccLogsQueryDefinition_logGroups
--- PASS: TestAccLogsSubscriptionFilter_Disappears_logGroup (788.10s)
=== CONT  TestAccLogsQueryDefinition_rename
--- PASS: TestAccLogsSubscriptionFilter_DestinationARN_kinesisStream (831.50s)
=== CONT  TestAccLogsQueryDefinition_disappears
--- PASS: TestAccLogsDataProtectionPolicy_basic (683.49s)
=== CONT  TestAccLogsMetricFilter_Disappears_logGroup
--- PASS: TestAccLogsSubscriptionFilter_basic (801.01s)
=== CONT  TestAccLogsMetricFilter_update
--- PASS: TestAccLogsDataProtectionPolicy_disappears (710.91s)
=== CONT  TestAccLogsMetricFilter_many
--- PASS: TestAccLogsQueryDefinition_disappears (165.28s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements
--- PASS: TestAccLogsSubscriptionFilter_disappears (817.62s)
=== CONT  TestAccLogsDataProtectionPolicyDocumentDataSource_empty
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_errorOnBadOrderOfStatements (19.41s)
=== CONT  TestAccLogsMetricFilter_disappears
--- PASS: TestAccLogsDestination_disappears (1079.68s)
=== CONT  TestAccLogsMetricFilter_basic
--- PASS: TestAccLogsDestination_basic (1093.94s)
=== CONT  TestAccLogsStream_disappears
--- PASS: TestAccLogsResourcePolicy_ignoreEquivalent (681.02s)
=== CONT  TestAccLogsGroup_namePrefix
--- PASS: TestAccLogsQueryDefinition_rename (332.27s)
=== CONT  TestAccLogsGroup_tags
--- PASS: TestAccLogsSubscriptionFilter_many (877.99s)
=== CONT  TestAccLogsGroup_disappears
--- PASS: TestAccLogsDataProtectionPolicy_policyDocument (1172.39s)
=== CONT  TestAccLogsStream_Disappears_logGroup
--- PASS: TestAccLogsMetricFilter_Disappears_logGroup (247.59s)
=== CONT  TestAccLogsGroup_nameGenerate
--- PASS: TestAccLogsSubscriptionFilter_distribution (1257.10s)
=== CONT  TestAccLogsGroup_basic
--- PASS: TestAccLogsGroup_disappears (139.11s)
--- PASS: TestAccLogsSubscriptionFilter_DestinationARN_kinesisDataFirehose (1265.65s)
--- PASS: TestAccLogsGroup_namePrefix (172.68s)
--- PASS: TestAccLogsMetricFilter_disappears (279.06s)
--- PASS: TestAccLogsQueryDefinition_logGroups (551.09s)
--- PASS: TestAccLogsResourcePolicy_basic (809.83s)
--- PASS: TestAccLogsMetricFilter_basic (257.65s)
--- PASS: TestAccLogsStream_disappears (249.56s)
--- PASS: TestAccLogsGroup_nameGenerate (139.49s)
--- PASS: TestAccLogsStream_Disappears_logGroup (190.98s)
--- PASS: TestAccLogsGroup_basic (127.77s)
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_empty (390.04s)
--- PASS: TestAccLogsMetricFilter_update (433.64s)
--- PASS: TestAccLogsSubscriptionFilter_roleARN (1405.10s)
--- PASS: TestAccLogsGroup_tags (305.70s)
--- PASS: TestAccLogsMetricFilter_many (451.40s)
--- PASS: TestAccLogsDataProtectionPolicyDocumentDataSource_basic (1454.96s)
--- PASS: TestAccLogsDestinationPolicy_basic (1575.79s)
--- PASS: TestAccLogsDestination_tags (1616.94s)
--- PASS: TestAccLogsDestination_update (1878.43s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/logs       1881.803s
```

File: internal/acctest/vcr.go (hunk 1 above)
Comment (ewbankkit) [reply internal/acctest/vcr.go:281]: I left it there as a reminder of what we want to do for VCR -- now that we have the API clients lazy loaded I _think_ we can eventually add this to the generated code.

**Comment by ewbankkit** [comment]
After `exportloopref` fixes:

```console
% make testacc TESTARGS='-run=TestAccLogsResourcePolicy_\|TestAccLogsQueryDefinition_' PKG=logs ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/logs/... -v -count 1 -parallel 3  -run=TestAccLogsResourcePolicy_\|TestAccLogsQueryDefinition_ -timeout 360m
=== RUN   TestAccLogsQueryDefinition_basic
=== PAUSE TestAccLogsQueryDefinition_basic
=== RUN   TestAccLogsQueryDefinition_disappears
=== PAUSE TestAccLogsQueryDefinition_disappears
=== RUN   TestAccLogsQueryDefinition_rename
=== PAUSE TestAccLogsQueryDefinition_rename
=== RUN   TestAccLogsQueryDefinition_logGroups
=== PAUSE TestAccLogsQueryDefinition_logGroups
=== RUN   TestAccLogsResourcePolicy_basic
=== PAUSE TestAccLogsResourcePolicy_basic
=== RUN   TestAccLogsResourcePolicy_ignoreEquivalent
=== PAUSE TestAccLogsResourcePolicy_ignoreEquivalent
=== CONT  TestAccLogsQueryDefinition_basic
=== CONT  TestAccLogsQueryDefinition_logGroups
=== CONT  TestAccLogsResourcePolicy_ignoreEquivalent
--- PASS: TestAccLogsQueryDefinition_basic (33.63s)
=== CONT  TestAccLogsResourcePolicy_basic
--- PASS: TestAccLogsResourcePolicy_ignoreEquivalent (47.24s)
=== CONT  TestAccLogsQueryDefinition_rename
--- PASS: TestAccLogsQueryDefinition_logGroups (55.72s)
=== CONT  TestAccLogsQueryDefinition_disappears
--- PASS: TestAccLogsQueryDefinition_disappears (27.98s)
--- PASS: TestAccLogsResourcePolicy_basic (56.39s)
--- PASS: TestAccLogsQueryDefinition_rename (55.01s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/logs	107.938s
```

File: internal/acctest/vcr.go (hunk 1 above)
Comment (jar-b) [reply internal/acctest/vcr.go:281]: Makes sense, thanks!

**Pull request merged**